src/algorithms/wolfe_line_search.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <RcppArmadillo.h>
+#include "../families/family.h"
+
+using namespace Rcpp;
+using namespace arma;
+
+template <typename T>
+double Family::zoom(const T& x, const mat&y, const double rho, const mat&u, 
```

Comment:
What does this function do? Can you add a comment describing the function or a more descriptive name?